### PR TITLE
[WIP]: PromptTemplate

### PR DESCRIPTION
**Pull request by EmanuelCampos**
DRAFT, not ready for use

missing:
- refactor missing modules
- refactor existent prompts
- tests with refine template
- fix circular dependency
- refactor partial format + prompt repack

### CODE REVIEW

File: packages/core/src/test.ts
Comment (EmanuelCampos) [inline packages/core/src/test.ts:1]: this file will be removed, just an example to test fast for now

File: packages/core/src/test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import {
+  ChatPromptTemplate,
+  PromptTemplate,
+  SimpleDirectoryReader,
+  VectorStoreIndex,
+  type ChatMessage,
+} from "./index.js";
+
+// Define a default prompt template
+const defaultTextQaPromptTemplate = () => `Context information is below.
```
Comment (marcusschiesser) [inline packages/core/src/test.ts:10]: does this need to be function?
```suggestion
const defaultTextQaPromptTemplate = "Context information is below.
```

File: packages/core/src/test.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const retriever = await index.asRetriever({});
+  retriever.similarityTopK = 10;
+
+  const queryEngine = index.asQueryEngine({
+    retriever,
+  });
+
+  // Update the prompts
+  queryEngine.updatePrompts({
+    "responseSynthesizer:textQATemplate": chatMessageTemplate,
```
Comment (marcusschiesser) [inline packages/core/src/test.ts:56]: can we directly pass the template?
```suggestion
    "responseSynthesizer:textQATemplate": [
  {
    content: "Always answer the question, even if you don't know the answer.",
    role: "system",
  },
  {
    content: defaultTextQaPromptTemplate(),
    role: "user",
  },
],
```
e.g. `updatePrompts` could generate a `ChatPromptTemplate` object if needed

Comment (marcusschiesser) [inline packages/core/src/test.ts:56]: How about adding a convenience function for the user that is updating the most likely prompt (here `responseSynthesizer:textQATemplate`):
```suggestion
  queryEngine.updatePrompt(chatMessageTemplate)
```

**Comment by EmanuelCampos** [comment]
closing, due to several changes on how it works, moving to https://github.com/run-llama/LlamaIndexTS/pull/665

@marcusschiesser thanks for the comments, appreciate your review on the new PR

**Pull request closed without merging**